### PR TITLE
[ODS-5729] Consider updating API error messages to be more meaningful to the API client 

### PR DESCRIPTION
**Pull request by mjaksn**
This updates the API error message referenced in the ticket to follow the proposed format. I searched for other error messages using the same form as the original message but found none.

### CODE REVIEW

**Comment by mjaksn** [comment]
Reopening this PR

**Comment by mjaksn** [comment]
These changes update the API error messages referenced in the ticket and another with the same form to be more concise and removes non-applicable wording. In addition, tests that include the affected error messages were also updated to align with the new language.

File: Application/EdFi.Ods.Api/Security/Authorization/AuthorizationBasisMetadataSelector.cs [hunk 1]
```diff
@@ -289,10 +289,9 @@ void ValidateAuthorizationContext(EdFiAuthorizationContext authorizationContext)
                     principal.Claims.SingleOrDefault(c => c.Type == EdFiOdsApiClaimTypes.ClaimSetName)?.Value;
 
                 response.SecurityExceptionMessage =
-                    $@"Access to the resource could not be authorized. Are you missing a claim? This resource can be authorized by the following claims:
-    {string.Join(Environment.NewLine + "    ", authorizingClaimNames)}
-The API client has been assigned the '{apiClientClaimSetName}' claim set with the following resource claims:
-    {string.Join(Environment.NewLine + "    ", apiClientResourceClaims)}";
+                    $@"Access to the resource could not be authorized. The API client has been assigned the '
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Api/Security/Authorization/AuthorizationBasisMetadataSelector.cs:292]: Current formatting is
```
Access to the resource could not be authorized. The API client has been assigned the '
            ClientClaimSet' claim set. Assign different claim set which includes one of the following claims to access this resource:
            claim one
    claim two
```

fix it to be like
```
Access to the resource could not be authorized. The API client has been assigned the 'ClientClaimSet' claim set.
Assign different claim set which includes one of the following claims to access this resource:
    claim one
    claim two
```

**Review by simpat-jesus — CHANGES_REQUESTED** [review]

File: Application/EdFi.Ods.Api/Security/Authorization/AuthorizationBasisMetadataSelector.cs (hunk 1 above)
Comment (simpat-jesus) [reply Application/EdFi.Ods.Api/Security/Authorization/AuthorizationBasisMetadataSelector.cs:292]: Even if the result is serialized to json, we are not sure how clients are parsing the message; so I think is better to return the message as readable as possible, even if that includes new line and carriage return characters once is serialized

Comment (mjaksn) [reply Application/EdFi.Ods.Api/Security/Authorization/AuthorizationBasisMetadataSelector.cs]: OK, I left them in and pushed the change.

**Review by simpat-jesus — APPROVED** [review]

**Pull request merged**